PR macports/macports-base#55 updates

slewsys opened the pull request:
Patches and updates for ports that conflict with macports/macports-base#55.

Updates verified as noted below:
FScript - Updated pmougin-F-Script-v2.1-0-g25c850c.zip checksums. Verify to patch stage only.
QMK-Groundstation - Requires QT 4.2.  Verify to patch stage only.
huevos - Source no longer distributed by author?  Verify to patch stage only.
libao - Verify to patch stage only.
daaplib - No longer distributed by author?  Build errors on macOS 13
git-sizer - Update to version 1.2.0 v. Installs and runs okay. 
play - Update to version 1.5. Installs okay.
starfighter - No longer distributed? Verify to patch stage only.
InsightToolkit* - InsightToolkit-devel (current version) has no conflicts.  
graphviz-oldgui - graphviz (current version) has no conflicts.
qmail-spamcontrol - Mark for removal.  Should be replaced by s/qmail.
MobileDevice - Installs okay.
ID3 - Installs okay.

macOS 10.13.4 17E202
Xcode 9.3 9E145

slewsys commented:
> On May 7, 2018, at 6:02 PM, Gerard <notifications@github.com> wrote:
> 
> @GerardSoleCa approved this pull request.
> 
> LGTM on git-sizer! Thanks 😄

Interesting app.  Ran it on macports-ports branch:

$ cd /ports
$ git-sizer
...
Processing blobs: 266253                        
Processing trees: 556318                        
Processing commits: 144138                        
Matching commits to trees: 144138                        
Processing annotated tags: 1                        
Processing references: 67                        
| Name                         | Value     | Level of concern               |
| ---------------------------- | --------- | ------------------------------ |
| Overall repository size      |           |                                |
| * Trees                      |           |                                |
|   * Total size               |  2.62 GiB | *                              |
|   * Total tree entries       |  74.9 M   | *                              |
|                              |           |                                |
| Biggest objects              |           |                                |
| * Commits                    |           |                                |
|   * Maximum size         [1] |  81.5 KiB | *                              |
| * Trees                      |           |                                |
|   * Maximum entries      [2] |  1.64 k   | *                              |
|                              |           |                                |
| Biggest checkouts            |           |                                |
| * Number of directories  [3] |  14.2 k   | *******                        |
| * Maximum path length    [4] |   139 B   | *                              |

[1]  794685b06f404f687b5d4eb6014229ade17017d1
[2]  2ef13d08f8b6b67133fac4d3232f60fb619d9fc4 (refs/heads/merged:perl)
[3]  4d7928db1b840ad0a7a754356132890cc26bef17 (747b2dec7d0e43dfa50469feb5831da51b6e560d^{tree})
[4]  69667f7e9a344efa59541a202d408fc65c35ba32 (766a7daf223f9113f65100007412589d680bd0da^{tree})

l2dy commented:
How is this related to #55, which is a ngircd update?

mojca commented:
It's related to https://github.com/macports/macports-base/pull/55, so it should also be referenced in a different way.

l2dy commented:
The commit messages should be edited to correct the reference. Use `https://github.com/macports/macports-base/pull/55` or `macports/macports-base#55` instead.

slewsys commented:
Mojca and Zero King, thank you for the clarification and edits.

slewsys commented:
Overlooked one update: ID3.

slewsys commented on aqua/FScript/Portfile:
```diff
@@ -28,15 +28,12 @@ master_sites            https://github.com/pmougin/F-Script/zipball/v${version}:
 distfiles               pmougin-F-Script-v2.1-0-g25c850c.zip:fscript
 
 checksums               pmougin-F-Script-v2.1-0-g25c850c.zip \
```
This may be a candidate for a [stealth update](https://trac.macports.org/wiki/PortfileRecipes#stealth-updates)?

pmetzger commented:
So at what point should this be merged? (Should it be merged now?)

slewsys commented:
On May 11, 2018, at 1:37 PM, Perry E. Metzger <notifications@github.com> wrote:
> 
> So at what point should this be merged? (Should it be merged now?)

We seem to be at loggerheads :) Ryan’s approval is contingent on changing the default behavior of fs-traverse per his review:  https://github.com/macports/macports-base/pull/55#issuecomment-387422633.  I countered Ryan’s request with an alternative: implement a new option to fs-traverse to alter its default behavior.

I’m happy to push either change or none at all.  So I defer to Ryan.
-AM

pmetzger commented:
@ryandesign As @slewsys has deferred to you, how do you want this handled?

ryandesign commented on audio/ID3/Portfile:
```diff
@@ -25,10 +25,6 @@ distname            ID3.framework.${version}
-post-extract {
-    move "${workpath}/id3 tag framework" ${worksrcpath}
-}
-
```
Committing this to ports now, without the changes from macports/macports-base#55, will cause the port to fail (`can't find file to patch at input line 3`). And of course even after the changes in macports/macports-base#55 are committed, users will not be running that code until after we release a new version of MacPorts and users have updated to it. So whatever changes are committed to ports for this issue (for all ports, not just this one) need to be made such that they work both with and without the changes from macports/macports-base#55. Compatibility code can be removed from Portfiles after all users have upgraded to the new version of MacPorts.

ryandesign commented on devel/daaplib/Portfile:
```diff
@@ -19,11 +19,8 @@ use_zip		yes
-	file rename ${workpath}/daaplib.${version}/daaplib/include \
-		${workpath}/include
-}
+distname        ${name}.${version}
```
With this change, the `distfiles` line is no longer necessary.

ryandesign commented on devel/git-sizer/Portfile:
```diff
@@ -3,7 +3,7 @@
 PortGroup               github 1.0
 
-github.setup            github git-sizer 1.1.0 v
+github.setup            github git-sizer 1.2.0 v
```
This does not appear to relate to this PR and should be done separately.

ryandesign commented on devel/play/Portfile:
```diff
@@ -4,7 +4,8 @@ PortSystem          1.0
 conflicts           sox
-version             1.2.7.2
+version             1.5.0
+revision            0
```
Does this update work without the base changes? If so, do it separately.

slewsys commented:
On May 11, 2018, at 4:30 PM, Ryan Schmidt <notifications@github.com> wrote:
> 
> @ryandesign commented on this pull request.
> 
> In aqua/FScript/Portfile:
> 
> > @@ -28,15 +28,12 @@ master_sites            https://github.com/pmougin/F-Script/zipball/v${version}:
>  distfiles               pmougin-F-Script-v2.1-0-g25c850c.zip:fscript
>  
>  checksums               pmougin-F-Script-v2.1-0-g25c850c.zip \
> 
> Yes, but it's also a candidate for update or deletion.

There is a new version of FScript on GitHub (https://github.com/adamnemecek/FScriptAnywhere) for recent versions of macOS (10.9+).  Older systems would need to use the existing port.  Though I’ve never had an opportunity to try this, it looks useful.  It's on my TODO list to create a port of the new FScript.

> In aqua/QMK-Groundstation/Portfile:
> 
> > @@ -46,7 +41,7 @@ post-build {
>  }
>  
>  destroot {
> -       	set appPath ${destroot}${applications_dir}
> -       	xinstall -d -m 755 ${appPath}
> -       	file copy ${worksrcpath}/build/Release/${name}.app ${appPath}/${name}.app
> +        set appPath ${destroot}${applications_dir}
> +        xinstall -d -m 755 ${appPath}
> +        file copy ${worksrcpath}/build/Release/${name}.app ${appPath}/${name}.app
> 
> This does not appear to relate to this PR. Whitespace-only changes should be done in a separate commit.

Whitespace patch has been reverted and squashed into original update.  Thank you for catching.

> 
> In audio/ID3/Portfile:
> 
> > @@ -25,10 +25,6 @@ distname            ID3.framework.${version}
>  checksums           rmd160  3b41296cca04d641c1e315fe7a93a0443ebbb8a5 \
>                      sha256  7d294b2348fc36ac8b068bab15e1b58602aafec098f670cccc52b7436b918270
>  
> -post-extract {
> -    move "${workpath}/id3 tag framework" ${worksrcpath}
> -}
> -
> 
> Committing this to ports now, without the changes from macports/macports-base#55, will cause the port to fail (can't find file to patch at input line 3). And of course even after the changes in macports/macports-base#55 are committed, users will not be running that code until after we release a new version of MacPorts and users have updated to it. So whatever changes are committed to ports for this issue (for all ports, not just this one) need to be made such that they work both with and without the changes from macports/macports-base#55. Compatibility code can be removed from Portfiles after all users have upgraded to the new version of MacPorts.

I haven’t tested, but I would expect most of these updates to not work without PR macports/macports-base#55.  

The best that I think of is associate PR macports/macports-base#55 with a major MacPorts version update and then condition the code on MacPorts' version.  Is it too late to consider PR macports/macports-base#55 for inclusion in MacPorts 2.5 release?  If so, perhaps there’s alternative for conditioning the code that I’m not aware of?

> In devel/daaplib/Portfile:
> 
> > @@ -19,11 +19,8 @@ use_zip		yes
>  distfiles	${name}.${version}${extract.suffix}
>  checksums	md5 038df4ac53780074e6e642a6e928f4fb
>  
> -post-extract	{
> -	file rename ${workpath}/daaplib.${version}/daaplib/src ${worksrcpath}
> -	file rename ${workpath}/daaplib.${version}/daaplib/include \
> -		${workpath}/include
> -}
> +distname        ${name}.${version}
> 
> With this change, the distfiles line is no longer necessary.

The distfiles line has been removed and squashed into original update. Again, thank you for catching.  


> In devel/git-sizer/Portfile:
> 
> > @@ -3,7 +3,7 @@
>  PortSystem              1.0
>  PortGroup               github 1.0
>  
> -github.setup            github git-sizer 1.1.0 v
> +github.setup            github git-sizer 1.2.0 v
> 
> This does not appear to relate to this PR and should be done separately.

You are correct - it does not depend on the base changes.  It has been removed and will be submitted separately.  

> In devel/play/Portfile:
> 
> > @@ -4,7 +4,8 @@ PortSystem          1.0
>  
>  name                play
>  conflicts           sox
> -version             1.2.7.2
> +version             1.5.0
> +revision            0
> 
> Does this update work without the base changes? If so, do it separately.

This update (in its current form) should not work without the base changes.

All of the issues that you caught should be addressed in the current updates, with the exception of backwards compatibility, which would seem to call for coordination with a major release.
-AM

ryandesign commented:
> There is a new version of FScript on GitHub (https://github.com/adamnemecek/FScriptAnywhere) for recent versions of macOS (10.9+).

As far as I can tell, that is not FScript; that is an Automator workflow for FScript. If you do have any information on updating FScript, please update the [ticket](https://trac.macports.org/ticket/44345).

> The best that I think of is associate PR macports/macports-base#55 with a major MacPorts version update and then condition the code on MacPorts' version.  Is it too late to consider PR macports/macports-base#55 for inclusion in MacPorts 2.5 release?  If so, perhaps there’s alternative for conditioning the code that I’m not aware of?

I think we should let this feature live on master for awhile before releasing it, in case any additional issues crop up.

The new code in macports/macports-base#55 runs before any port `post-extract` blocks, right? So port `post-extract` blocks that rename the worksrcdir could be updated to first check whether the directory already exists at the new name. If so, we must be running on a new version of base and there's no need for the port to rename it.

ryandesign commented:
> The new code in macports/macports-base#55 runs before any port post-extract blocks, right? So port post-extract blocks that rename the worksrcdir could be updated to first check whether the directory already exists at the new name. If so, we must be running on a new version of base and there's no need for the port to rename it.

Like what the MobileDevice port was doing.

If we do it this way, it's probably helpful to add a comment mentioning this PR or macports/macports-base#55, and if we already know what needs to be done after the new version of base with that PR merged (i.e. the `post-extract` block can be removed) we could mention that in the comment too.

slewsys commented:
> On May 11, 2018, at 9:06 PM, Ryan Schmidt <notifications@github.com> wrote:
> 
> As far as I can tell, that is not FScript; that is an Automator workflow for FScript. If you do have any information on updating FScript, please update the ticket.

That may be.  I’ll look into it further and update the ticket if I find something.

>> The best that I think of is associate PR macports/macports-base#55 with a major MacPorts version update and then condition the code on MacPorts' version. Is it too late to consider PR macports/macports-base#55 for inclusion in MacPorts 2.5 release? If so, perhaps there’s alternative for conditioning the code that I’m not aware of?
> 
> I think we should let this feature live on master for awhile before releasing it, in case any additional issues crop up.

Okay.

> The new code in macports/macports-base#55 runs before any port post-extract blocks, right?

Right.

> So port post-extract blocks that rename the worksrcdir could be updated to first check whether the directory already exists at the new name. If so, we must be running on a new version of base and there's no need for the port to rename it.

Good idea.  Looking at the updates, I think that should work.

> Like what the MobileDevice port was doing.
> 
> If we do it this way, it's probably helpful to add a comment mentioning this PR or macports/macports-base#55, and if we already know what needs to be done after the new version of base with that PR merged (i.e. the post-extract block can be removed) we could mention that in the comment too.

How about:

This code block is provided for backward compatibility with MacPorts base versions prior to PR macports/macports-base#55. If backward compatibility is no longer needed, then this code can be removed.

Thank you for your help!
-AM

slewsys commented:
For explanation of `octave-1.0.tcl` port group update, please refer to comment https://github.com/macports/macports-base/pull/55#issuecomment-389115385.

pmetzger commented:
Okay, what do we need to do next in order to merge this?

slewsys commented:
On May 15, 2018, at 8:49 AM, Perry E. Metzger <notifications@github.com> wrote:
> 
> Okay, what do we need to do next in order to merge this?

Give everyone a little time to digest the proposed changes and offer feedback.  It looks good to me, but there’s a lot of nuance in MacPorts, and I might be missing something.
-AM

pmetzger commented:
Shall we pick a date for it? Maybe Thursday, May 17?

raimue commented:
The change in macports/macports-base#55 was not even merged yet. These are changes for functionality that is not available in any version of MacPorts, not even master. Why would we want to merge this at all right now?

pmetzger commented:
@raimue I'm mostly just trying to figure out what the steps are before this gets merged, who has to take those steps, and when they're likely to happen. I am happy regardless of the answers to that so long as they're concrete. That way I know what I have to look for to get something shepherded towards completion.

slewsys commented:
On May 16, 2018, at 11:00 AM, Rainer Müller <notifications@github.com> wrote:
> 
> The change in macports/macports-base#55 was not even merged yet. These are changes for functionality that is not available in any version of MacPorts, not even master. Why would we want to merge this at all right now?

Ryan requested that all updates in #1760 (macports/macports-ports#1760) be made backward compatible with MacPorts base versions prior to any changes in macports/macports-base#55.  That should be the case now, barring oversights on my parts.  So it should be safe to merge #1760 updates without merging macports/macports-base#55.

One caveat is that #1760 includes an update to the octave port group, so I was hoping MarcusCalhoun-Lopez, or whoever’s maintaining that group, could sign off on the update prior to commit.

I did have to update the octave port in order to get it to install on my system. PRs are being prepared for that separately.
-AM

raimue commented:
Yes, it would be safe to merge #1760 now, but also completely unnecessary. Unless we are also going to merge macports/macports-base#55, no changes in ports are required. The discussion in that pull request is still ongoing and I do not see we came to a conclusion there.

In my opinion we can fix the ports after macports/macports-base#55 has been merged, as long as we do it before the 2.6.0 release this will be part of. Users following master have to expect breakage from time to time and there is no need to put in changes to ports before the feature has even landed on master.

slewsys commented:
On May 16, 2018, at 4:29 PM, Rainer Müller <notifications@github.com> wrote:
> 
> Yes, it would be safe to merge #1760 now, but also completely unnecessary. Unless we are also going to merge macports/macports-base#55, no changes in ports are required. The discussion in that pull request is still ongoing and I do not see we came to a conclusion there.
> 
> In my opinion we can fix the ports after macports/macports-base#55 has been merged, as long as we do it before the 2.6.0 release this will be part of. Users following master have to expect breakage from time to time and there is no need to put in changes to ports before the feature has even landed on master.

That’s fine.  Just wanted to clarify that PR #1760 is a non-breaking update, unlike macports/macports-base#55.

I had overlooked the base/src/pextlib1.0 tests.  Looking at those now.  Thank you for the heads up!
-AM

pmetzger commented:
@MarcusCalhoun-Lopez might not know that he was invoked in this discussion.

pmetzger commented:
So where are we on this (and on https://github.com/macports/macports-base/pull/55 ?)

pmetzger commented:
@slewsys @raimue Poke?

pmetzger commented:
@raimue ?

slewsys commented:
On Jul 25, 2018, at 4:02 PM, Perry E. Metzger <notifications@github.com> wrote:
> 
> @raimue ?

The crux of the problem, I think, is that either every port group adopts the ability the handle multiple (and misnamed) extracted files, or a rule is adopted where only a single file is extracted in workdir.  So unless this issue is likely to be resolved one way or another any time soon, I would argue that committing PR macports/macports-base#55 won’t meet much resistance.

For the record, one way of enforcing a single extraction would be to bundle multiple downloads into a tar archive of a subdirectory.
-AM

pmetzger commented:
I note this has been sitting uncommitted for a very long time. What should we do next to move it forward towards resolution?

pmetzger commented:
@raimue @ryandesign @mojca Any ideas on how to move this forward? Or should it be closed for now as something no one wants to complete?

slewsys commented:
On Oct 1, 2018, at 6:40 PM, Perry E. Metzger <notifications@github.com> wrote:
> 
> @raimue @ryandesign @mojca Any ideas on how to move this forward? Or should it be closed for now as something no one wants to complete?


Just commit it.  It doesn’t bite...

mojca commented:
I initially thought that the PR from base needs to be committed first, but looking at the changes again, it mostly looks ok (I did not test though). Three ports failed to build by Travis, logs are no longer accessible, but I suspect these ports probably did not build earlier either (again, I did not test).

@raimue questioned the necessity of this commit if we never get around to committing macports/macports-base#55. On one hand this is true: until that change is made, this PR is in theory useless, but at the same time also completely harmless. I would say that having it linger here for another couple of months is also completely useless and wasting precious resources (like @pmetzger and potentially others spending time on it when other work could be done :). And if we just close it and reopen it, there might be further conflicts that will need resolving, so again more work.

I would like to see that PR in base committed for the sake of simplifying many Portfiles. So I would vote for merging this one also. If we eventually implement a different solution in base ... well, we can always revert these commits.

mojca commented:
My only criticism would be that some commit messages are too long :) :) :)

raimue commented:
I still think this is blocked by macports/macports-base#55 (does not make sense without having the change in base first) and over there we are waiting for a cleanup of the commits. At least that is how I understood the latest status. Am I mistaken?

mojca commented:
I can take a short look at FScript (@ci42) to see if it's possible to fix it in a slightly better way.

pmetzger commented:
@mojca Seems good. Do it!

pmetzger commented:
FYI, https://github.com/macports/macports-base/pull/55 has now been merged.

pmetzger commented:
Okay, just to poke everyone again: https://github.com/macports/macports-base/pull/55 has been merged, so this can now go in, though it needs to be updated. It's the oldest pull request in the system, dating from May(!!), and it would be nice to get it resolved finally.

slewsys commented:
> On Nov 5, 2018, at 1:23 PM, Mojca Miklavec <notifications@github.com> wrote:
> 
> @mojca commented on this pull request.
> 
> Apart from resolving the conflict I would try to honour @ryandesign's request to split the port update into a separate PR/commit, independent of this one.
> 
> In aqua/QMK-Groundstation/Portfile:
> 
> >  post-extract {
> -    # rename worksrcdir to remove a space
> -    file rename "${workpath}/${name} ${version}" ${workpath}/${worksrcdir}
> +	if {![file exists ${worksrcpath}]} {
> +		file rename "${workpath}/${name} ${version}" ${worksrcpath}
> +	}
> No tabs, please.
> 

Hi Mojca,
I'm unable to submit any code changes until later this month. I’m not sure which request you're referring to. Do to PR need to be reorganized? 
-AM

mojca commented:
I would say that this PR just needs the following (relatively trivial/straightforward) changes.
* Create a separate PR or a separate commit (or just commit directly) the update for the `play` port (@ci42). Independent of this ticket.
* "Redo" the work for the `daaplib` port which has been modified quite a bit by @ryandesign in the meantime.
* Replace some accidental tabs with spaces in `QMK-Groundstation`.
* (This is just a matter of taste and thus fully optional. The number of remaining changes is so small and uniform that the commits could potentially be squashed. I would probably remove `Prepare for PR macports/macports-base#55.` from the first line of commit message and replace it with some meaniningful description about what exactly was done, something like `optionally rename worksrcdir`, but I'm sure there's a better way to say it. And then move `See: https://github.com/macports/macports-base/pull/55` to the 3rd line of commit message.)
Apart from those minor changes it looks fine to commit.

Anyone of us could do those changes as well. In fact we need to commit them relatively soon, else those ports will be broken for anyone using the `master` branch (probably a small number of experienced developers only who most likely don't even use any of those random ports anyway, but still).

ryandesign commented:
> * "Redo" the work for the `daaplib` port which has been modified quite a bit by @ryandesign in the meantime.

No further changes are required for daaplib in relation to this PR; that portion of this PR can be deleted. Sorry, I didn't realize/remember there were changes for daaplib in this PR when I made my separate changes.

mojca commented:
Regarding the `QMK-Groundstation`:
* To start with, it did not even fetch correctly for me. I had to change the server to `http://svn.mikrokopter.de/websvn/filedetails.php?repname=Projects&path=%2FQMK-Groundstation%2FRelease%2F`.
* `sha256` and livecheck are missing.
* I was trying to figure out whether `worksrcdir "${name} ${version}"` would work, but it doesn't come anywhere close to building.

pmetzger commented:
@mojca Any port that has been broken long enough without anyone complaining is probably not worth keeping in the tree... :(

mojca commented:
Ad `MobileDevice`: the build systems seems quite a bit broken. It originally installs `Distutils` egg instead of `MobileDevice`. And it requires the top folder to be called `MobileDevice` and nothing else: it attempts to package `../MobileDevice` instead of having that folder one level deeper. This is something I would attempt to fix by moving the files to a subfolder rather than renaming the whole directory, but ...

The port has been added in 2014 (https://trac.macports.org/ticket/42596), the one person asking for it never contributed any further and has been removed as maintainer in March this year.
@neverpanic filed 3 bug reports upstream for trivial issues, none of which even got any acknowledgement in 4 years. There have never been any releases, just 13 commits in the last 4 years, nobody asked for updates. I wonder if we need that port.


The remaining ports are either those with some spaces in names, and one from gitlab (apparently with the same needs as github which already handles this correctly, except that there's no gitlab portgroup to do it in the port's name).

mojca commented:
Ad `Huevos`: indeed, the sources (not even binaries) are no longer available upstream, and on top of that there's a build failure:
```
xcodebuild: error: Unable to read project 'Huevos.pbproj'.
	Reason: Project file 'Huevos' was written by an unsupported Xcode version (38)
```
so we may just as well delete that port as well.

pmetzger commented:
@mojca I fully support removing these. I'd suggest you file distinct PRs to remove each of those ports, we can give people a bit of time to object in each case, and then we can merge them.

slewsys commented:
Following Mojca's lead, PRs have been removed for:
```  
QMK-Groundstation
huevos
starfighter
MobileDevice
```

in favor of deleting these ports.

I also propose deleting port:
```
play
```

until someone is prepared to fix this.  The problem is that the current version does not build.  And while updating to version 1.5.2 (the current release) works, it will require some additional packaging to install in Macports' framework.  In particular, the python script `play` used to create a new project depends upon python libraries for which no installer is available (see _framework/pym_ in the distribution).

Consequently, the PR to update `play` has been omitted as well.

slewsys commented:
Just to clarify, the current version of `play` is actually 2.6.20, and built with a Scala compiler (`sbt`).  More info [here](https://www.playframework.com).

pmetzger commented:
@mojca: thoughts?

pmetzger commented:
I'd like to suggest that 

1. This PR be closed
2. @slewsys open new individual PRs for any ports where this remains relevant.

@slewsys ? @mojca ?

slewsys commented:
@pmetzger @mojca There remain several outstanding issues:

1. Submit PR for libao with remaining patch in this PR.
2. Submit delete requests for:

- QMK-Groundstation
- huevos
- starfighter
- MobileDevice
- play (update to version 2.6.20 requires rewriting Portfile)

3. Submit PR for oxygen, which silently fails when scanning root tree which is a symlink as reported by @eborisch [here](https://github.com/macports/macports-base/pull/55#issuecomment-443875822).

Regarding the third issue, a cursory look at the code suggests this would take a weekend which I don't have presently to write a fix.

So I can submit a PR for libao.  Do I need to submit delete requests for the other ports?

eborisch commented:
RE #3: nine year-old PR exists upstream: https://github.com/doxygen/doxygen/issues/3518

pmetzger commented:
Okay, my intention is to close this at some point in the next week or so. I would recommend that we get the new PRs in place before then. This is by far the oldest PR alive in the system and I'd like it put to bed finally.

slewsys commented:
I’ll get to it.  Thank you for the heads up.
-AM

> On Dec 19, 2018, at 9:15 AM, Perry E. Metzger <notifications@github.com> wrote:
> 
> Okay, my intention is to close this at some point in the next week or so. I would recommend that we get the new PRs in place before then. This is by far the oldest PR alive in the system and I'd like it put to bed finally.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

mojca commented:
And #2951.
